src/app/panels/righthand-panel.js

```diff
@@ -40,27 +42,37 @@ function RighthandPanel (appAPI, events, opts) {
     this.event.trigger('switchTab', [tabClass])
   }
 
-  // load tabbed menu component
-  var tabEvents = {compiler: events.compiler, app: events.app, rhp: self.event}
-  self._view.tabbedMenu = new TabbedMenu(options, tabEvents)
-
   events.rhp = self.event
 
-  this._view.tabbedMenu.addTab('Compile', 'compileView', compileTab(optionViews, appAPI, events, opts))
-  this._view.tabbedMenu.addTab('Run', 'runView', runTab(optionViews, appAPI, events))
-  this._view.tabbedMenu.addTab('Settings', 'settingsView', settingsTab(optionViews, appAPI, events))
-  this._view.tabbedMenu.addTab('Analysis', 'staticanalysisView', analysisTab(optionViews))
-  this._view.tabbedMenu.addTab('Debugger', 'debugView', debuggerTab(optionViews))
-  this._view.tabbedMenu.addTab('Support', 'supportView', supportTab(optionViews, events))
+  var compileTab = new CompileTab(appAPI, events)
+  optionViews.appendChild(compileTab.render())
+  var runTab = new RunTab(appAPI, events)
+  optionViews.appendChild(runTab.render())
+  var settingsTab = new SettingsTab(appAPI, events)
+  optionViews.appendChild(settingsTab.render())
+  var analysisTab = new AnalysisTab(appAPI, events)
+  optionViews.appendChild(analysisTab.render())
+  var debuggerTab = new DebuggerTab(appAPI, events)
+  optionViews.appendChild(debuggerTab.render())
+  var supportTab = new SupportTab(appAPI, events)
+  optionViews.appendChild(supportTab.render())
+  this._view.tabbedMenu.addTab('Compile', 'compileView', compileTab.render())
+  this._view.tabbedMenu.addTab('Run', 'runView', runTab.render())
+  this._view.tabbedMenu.addTab('Settings', 'settingsView', settingsTab.render())
+  this._view.tabbedMenu.addTab('Analysis', 'staticanalysisView', analysisTab.render())
+  this._view.tabbedMenu.addTab('Debugger', 'debugView', debuggerTab.render())
+  this._view.tabbedMenu.addTab('Support', 'supportView', supportTab.render())
   this._view.tabbedMenu.selectTabByTitle('Compile')
 
   self.pluginManager = new PluginManager(appAPI, events)
   events.rhp.register('plugin-loadRequest', (json) => {
-    var content = pluginTab(optionViews, json.url)
+    var content = new PluginTab(appAPI, events, json)
     this._view.tabbedMenu.addTab(json.title, 'plugin', content)
     self.pluginManager.register(json, content)
   })
 
+  self.showDebugger = function () { appAPI.switchTab('debugView') }
```

Comment:
if we rollback and keep using ``debuggingRequested`, I think this ^ is not needed anymore

Reply:
ok